src/chromo/decay_handler.py

```diff
@@ -0,0 +1,204 @@
+        # Store stable and decaying PDG IDs for future reference
+        all_stable_pids = []
+        all_decaying_pids = []
+        for pentry in self.pythia.particleData.all():
+            if pentry.mayDecay and pentry.canDecay:
+                if pentry.hasAnti:
+                    all_decaying_pids.append(pentry.antiId)
+                all_decaying_pids.append(pentry.id)
+            else:
+                if pentry.hasAnti:
+                    all_stable_pids.append(pentry.antiId)
+                all_stable_pids.append(pentry.id)
+
+        self.all_decaying_pids = np.array(all_decaying_pids, dtype=np.int64)
+        self.all_stable_pids = np.array(all_stable_pids, dtype=np.int64)
+
+    def _set_exclusive_stability(self, pids, is_decaying):
+        """
+        Set particles exclusively as stable or decaying based on the provided list.
+
+        Parameters:
+            pids (list[int]): List of PDG IDs for particles to be exclusively set.
+            is_decaying (bool): True if the particles should be set as decaying,
+            False for stable.
+
+        Raises:
+            ValueError: If exclusivity conditions are not met.
+        """
+        if (
+            sum(
+                x is not None
+                for x in [
+                    self.extra_stable_pids,
+                    self.extra_decaying_pids,
+                    self.stable_pids,
+                    self.decaying_pids,
+                ]
+            )
+            != 1
+        ):
+            raise ValueError(
+                "stable_pids or decaying_pids can be set "
+                "if all other options are None"
+            )
+
+        # Set particles exclusively as stable or decaying
+        for pentry in self.pythia.particleData.all():
+            self.pythia.particleData.mayDecay(pentry.id, not is_decaying)
+        for pid in pids:
+            self.pythia.particleData.mayDecay(pid, is_decaying)
+
+    def __call__(self, event):
+        """
+        Decay particles in the provided `event` that are marked as `decaying`.
+
+        After the call, `event` contains both the initial particles and
+        the newly produced particles through decay.
+        The decayed particles have their `event.status` set to 2.
+
+        Parameters:
+            event (Event): The event containing particles to be decayed.
+        """
+        # Return if the event is empty
+        if len(event.pid) == 0:
+            return
+
+        # Save the original status of non-final particles before decay
+        # as they will become 2
+        not_final = event.status != 1
+        status = np.copy(event.status[not_final])
+
+        # Particles with negative status are not processed by Pythia
+        event.status[not_final] = -1
+
+        # Save refs to original event parameters
+        # that are not passed to Pythia8 stack
+        charge = event.charge
+        vx = event.vx
+        vy = event.vy
+        vz = event.vz
+        vt = event.vt
```

Comment:
This is not making copies. Is this code correct?

Reply:
Turns out that this was correct, since you are replacing the arrays instead over overriding their contents later.